### PR TITLE
Use source mass flow rate in unmet energy calculation on stratified tank

### LINKED ISSUE #7275: Stratified tank source heat transfer using use flow rate instead of source
Issue overview
--------------
A very astute observer of the E+ source code noticed that we are [using the use flow rate in the source side energy calculation](https://github.com/NREL/EnergyPlus/blob/1afc16dac65705550eb9e08c5398d29f434f41ee/src/EnergyPlus/WaterThermalTanks.cc#L8294). Should be easy enough to fix. 

### Details
Some additional details for this issue (if relevant):
 - Version of EnergyPlus : 9.1

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by nmerket**
fixes #7275

Pull request overview
---------------------
This addresses an issue in the stratified tank where the use mass flow rate was used when the source mass flow rate should have been used. Fortunately, this doesn't affect the energy balance on the tank since the variable in question is only used for the unmet energy/rate reporting variable calculation. 

I branched from 9.1 in case this could go into a bugfix release. But given that it's a minor issue that only affects a single reporting variable it's not terribly urgent to me.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by nrel-bot-2** [comment]
@nmerket it has been 22 days since this pull request was last updated.

**Comment by nmerket** [comment]
In my run of the regression tests, I saw one file with big mathdiffs: `WaterHeaterStandAlone`. These are the only diffs in that file:

stratified_tank_unmet_demand

This is to be expected because the correct flow rate is being used to calculate those unmet loads outputs now, which was the point of this bugfix. 

In my runs, I'm also seeing a few table diffs, but they're all the annual heat emissions and the difference is in the 10th significant digit. 

We'll see if CI turns up the same things.

**Comment by nmerket** [comment]
Just looked at the CI results, it looks like the table diffs only showed up in my runs and the only difference was the ESO diffs plotted above.

**Comment by mjwitte** [comment]
@nmerket CI only runs design days which supresses ABUPS and many of the table outputs, so those diffs don't come up on CI.

File: src/EnergyPlus/WaterThermalTanks.cc [hunk 1]
```diff
@@ -8291,7 +8291,7 @@ namespace WaterThermalTanks {
                             return Qheatpump * node.HPWHWrappedCondenserHeatingFrac;
                         }
                     } else {
-                        return node.UseMassFlowRate * Cp * (Tank.SourceInletTemp - Tavg[i]);
+                        return node.SourceMassFlowRate * Cp * (Tank.SourceInletTemp - Tavg[i]);
```
Comment (Myoldmopar) [inline src/EnergyPlus/WaterThermalTanks.cc:8294]: Yeah, I mean this looks much better.

File: tst/EnergyPlus/unit/WaterThermalTanks.unit.cc [hunk 2]
```diff
@@ -2051,5 +2051,134 @@ TEST_F(EnergyPlusFixture, StratifiedTankCalc)
...
+
+    WaterThermalTanks::CalcWaterThermalTankStratified(TankNum);
+
+    Real64 EnergySum = 0.0;
+    for (int i = 0; i < Tank.Nodes; ++i) {
+        auto &node = Tank.Node[i];
+        EnergySum += node.Mass * Cp * (node.Temp - 60.0);
+    }
+    Real64 Esource = Tank.SourceEffectiveness * Tank.SourceMassFlowRate * Cp * (Tank.SourceInletTemp - Tank.Node(Tank.SourceOutletStratNode).TempAvg) * TimeStepSys * SecInHour;
+    EXPECT_NEAR(Esource, EnergySum, EnergySum * 0.001);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/WaterThermalTanks.unit.cc:2181]: Very nice unit test :+1:

**Review by Myoldmopar — APPROVED** [review]

**Pull request merged**